### PR TITLE
gitlab: 16.1.3 -> 16.3.0

### PR DESCRIPTION
**Pull request by ajs124**
###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yayayayaka** [comment]
Result of `nixpkgs-review pr 245517` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>gitaly</li>
    <li>gitlab</li>
    <li>gitlab-container-registry</li>
    <li>gitlab-ee</li>
    <li>gitlab-elasticsearch-indexer</li>
    <li>gitlab-pages</li>
    <li>gitlab-workhorse</li>
  </ul>
</details>

**Review by yayayayaka — COMMENTED** [review]
Cherry-picked to a release-23.05 branch and deployed for manual testing. Turns out, the `gitlab-db-config.service` keeps failing on my end and it has been [reported to gitlab already](https://gitlab.com/gitlab-org/gitlab/-/issues/419593).

Also worth noting:
```
Jul 26 16:02:43 services y62wgksq521yrmnkg8i48zdvxhpflsm5-gitlab-db-config[231061]: WARNING: The Rust extension for prometheus-client-mmap is unavailable, falling back to the legacy C extension.
Jul 26 16:02:43 services y62wgksq521yrmnkg8i48zdvxhpflsm5-gitlab-db-config[231061]:   The Rust extension will be required in the next version. If you are compiling this gem from source,
Jul 26 16:02:43 services y62wgksq521yrmnkg8i48zdvxhpflsm5-gitlab-db-config[231061]:   ensure your build system has a Rust compiler and clang: https://gitlab.com/gitlab-org/ruby/gems/prometheus-client-mmap
```

**Comment by ajs124** [comment]
> Cherry-picked to a release-23.05 branch and deployed for manual testing. Turns out, the `gitlab-db-config.service` keeps failing on my end and it has been [reported to gitlab already](https://gitlab.com/gitlab-org/gitlab/-/issues/419593).

Interesting. I didn't see that, when upgrading.

I also saw the rust thing, but thought that would be a future me/us problem :grimacing:

File: pkgs/applications/version-management/gitlab/gitlab-elasticsearch-indexer/default.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
 
 buildGoModule rec {
   pname = "gitlab-elasticsearch-indexer";
-  version = "4.3.5";
+  version = "4.3.6";
```
[the remark this answers is not in the document]
Comment (ajs124) [reply pkgs/applications/version-management/gitlab/gitlab-elasticsearch-indexer/default.nix:5]: can you push to this branch? if so, feel free to ad that change.

[the remark this answers is not in the document]
Comment (ajs124) [reply pkgs/applications/version-management/gitlab/gitlab-elasticsearch-indexer/default.nix:5]: pushed it with the update to 16.2.2

**Comment by yayayayaka** [comment]
Marking as draft until [upstream](https://gitlab.com/gitlab-org/gitlab/-/issues/419593) comes up with a solution as it seems to affect a wide range of installations.

**Comment by yayayayaka** [comment]
So apparently all users who created or upgraded their GitLab via versions 15.11.0 - 15.11.2 will run into the aforementioned issue. Their solution will be to [just let the user clean up the mess](https://gitlab.com/gitlab-org/gitlab/-/blob/cf2fb5b9f13dce65260f49dd5f40a2d9eb02abf1/doc/update/versions/gitlab_16_changes.md#undefined-column-error-upgrading-to-162-or-later).

Unsure how to proceed with this. Maybe add a warning informing possible affected users to apply the suggested migrations manually and mark this PR as ready again?

**Comment by yayayayaka** [comment]
Result of `nixpkgs-review pr 245517` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>gitaly</li>
    <li>gitlab</li>
    <li>gitlab-container-registry</li>
    <li>gitlab-ee</li>
    <li>gitlab-elasticsearch-indexer</li>
    <li>gitlab-pages</li>
    <li>gitlab-shell</li>
    <li>gitlab-workhorse</li>
  </ul>
</details>

**Pull request closed without merging**